capstone-project/src/main/angular-webapp/src/app/map/map.component.ts

```diff
@@ -40,24 +41,43 @@ export class MapComponent implements OnInit {
       .toPromise()
       .then((response) => {
         for (let key in response) {
-          this.addMarkerForDisplay(response[key]);
+
+          // If the marker has a blob key - get its URL 
+          if (response[key].blobKey) {
+            let imageUrl;
+            this.httpClient.get('/blob-service?' + 'blobAction=' + BlobAction.KEY_TO_BLOB +  '&blob-key=' + response[key].blobKey, { responseType: 'blob' })
+              .toPromise()
+              .then((blob) => {
+                imageUrl = MapComponent.getUrlFromBlob(blob)
+                this.addMarkerForDisplay(response[key], imageUrl)
+              });
+          }
+          else {
+            this.addMarkerForDisplay(response[key]);
+          }
```

Comment:
Currently, this isn't an issue (because we have a few pictures), but, consider lazy loading those. 

i.e only when the marker is actually being clicked. We can talk about it more in today's meeting